sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/rx/QueryValidationTests.java

```diff
@@ -317,6 +325,132 @@ public void queryPlanCacheSinglePartitionParameterizedQueriesCorrectness() {
+
+        //Insert some documents
+        List<TestObject> testObjects = insertDocuments(itemCount, Arrays.asList("CA", "US"), container);
+
+        List<String> sortedObjects = testObjects.stream()
+                                         .sorted(Comparator.comparing(TestObject::getProp))
+                                         .map(TestObject::getId)
+                                         .collect(Collectors.toList());
+
+        String query = "Select * from c";
+        String orderByQuery = "select * from c order by c.prop";
+
+        List<PartitionKeyRange> partitionKeyRanges = getPartitionKeyRanges(containerId, asyncDocumentClient);
+        String requestContinuation = null;
+        String orderByRequestContinuation = null;
+        int preferredPageSize = 15;
+        ArrayList<TestObject> resultList = new ArrayList<>();
+        ArrayList<TestObject> orderByResultList = new ArrayList<>();
+
+        // Query
+        FeedResponse<TestObject> jsonNodeFeedResponse = container
+                                                            .queryItems(query, new CosmosQueryRequestOptions(), TestObject.class)
+                                                            .byPage(preferredPageSize).blockFirst();
+        assert jsonNodeFeedResponse != null;
+        resultList.addAll(jsonNodeFeedResponse.getResults());
+        requestContinuation = jsonNodeFeedResponse.getContinuationToken();
+
+        // Orderby query
+        FeedResponse<TestObject> orderByFeedResponse = container
+                                                           .queryItems(orderByQuery, new CosmosQueryRequestOptions(),
+                                                                       TestObject.class)
+                                                           .byPage(preferredPageSize).blockFirst();
+        assert orderByFeedResponse != null;
+        orderByResultList.addAll(orderByFeedResponse.getResults());
+        orderByRequestContinuation = orderByFeedResponse.getContinuationToken();
+
+        // Scale up the throughput for a split
+        logger.info("Scaling up throughput for split");
+        ThroughputProperties throughputProperties = ThroughputProperties.createManualThroughput(16000);
+        ThroughputResponse throughputResponse = container.replaceThroughput(throughputProperties).block();
+        logger.info("Throughput replace request submitted for {} ",
+                    throughputResponse.getProperties().getManualThroughput());
+        throughputResponse = container.readThroughput().block();
+
+
+        // Wait for the throughput update to complete so that we get the partition split
+        while (true) {
+            assert throughputResponse != null;
+            if (!throughputResponse.isReplacePending()) {
+                break;
+            }
+            logger.info("Waiting for split to complete");
+            try {
+                Thread.sleep(10 * 1000);
+            } catch (InterruptedException e) {
```

Comment:
we shouldn't catch InterruptedEXception. if this is thrown it should naturally kill the test.

Reply:
Done